kratos/python_scripts/file_points_output_process.py

```diff
@@ -0,0 +1,118 @@
+from __future__ import print_function, absolute_import, division  # makes KratosMultiphysics backward compatible with python 2.6 and 2.7
+import csv, os
+
+# Importing the Kratos Library
+import KratosMultiphysics
+
+# other imports
+from KratosMultiphysics.multiple_points_output_process import MultiplePointsOutputProcess
+
+def Factory(settings, Model):
+    if(type(settings) != KratosMultiphysics.Parameters):
+        raise Exception("expected input shall be a Parameters object, encapsulating a json string")
+    return FilePointsOutputProcess(Model, settings["Parameters"])
+
+class FilePointsOutputProcess(KratosMultiphysics.Process):
+    """This process writes output for several points given in a file to an output file
+    Internally it holds an object of type "MultiplePointsOutputProcess"
+    Usage:
+        - file path of an .csv-file (!) for the import of sampling points need to be specified
+        - the first line of the .csv-file should state the column names: "x,y,z"
+        - the following rows should each be one point with its x,y and z value in the respective column of the .csv-file, separated by a comma 
+        - the number of sampling points needs to be specified
+    """
+    def __init__(self, model, params):
+        KratosMultiphysics.Process.__init__(self)
+
+        default_settings = KratosMultiphysics.Parameters('''{
+            "help"              : "This process writes output for several points given in a file to a file. Internally it holds an object of type MultiplePointsOutputProcess",
+            "model_part_name"   : "",
+            "entity_type"       : "element",
+            "file_path"         : "",
+            "sampling_points"   : 3,
+            "output_variables"  : [],
+            "historical_value"  : true,
+            "search_tolerance"  : 1e-6,
+            "print_format"      : "",
+            "output_file_settings": {}
+        }''')
+
+        params.ValidateAndAssignDefaults(default_settings)
+
+        # retrieving file path and name
+        file_path = params["file_path"].GetString()
+        if len(file_path) == 0:
+            raise Exception('A file path has has to be provided!')
+
+        # get and check number of sampling points
+        number_of_sampling_points = params["sampling_points"].GetInt()
+        if number_of_sampling_points <= 0:
+            raise Exception('The number of sampling points has to be larger than 0!')
+
+        # check the entity type
+        entity_type = params["entity_type"].GetString()
+        if not entity_type == "element" or entity_type == "condition":
+            raise Exception('"entity_type" can be either "element" or "condition"!')
+
+        # initialize position matrix for sampling points
+        positions = KratosMultiphysics.Matrix(number_of_sampling_points, 3)
+
+        # check and read file
+        if os.path.exists(file_path) and os.access(file_path, os.R_OK):
```

Comment:
this check is not necessary, just leave this to the basic implementation / the OS

Reply:
I thought it might help the user, because it helped me while debugging, since there is a difference between "there is something wrong with the file" (e.g. wrong header, not .csv, ..) and "the file can not be found where it is supposed to be".

Reply:
I would completely remove the try-except and let the code tell you what is wrong
e.g. if it cannot find the file it will throw in the line `with open(file_path, newline=None) as file:`

if there is sth wrong with the csv file it wll throw in line `reader = csv.DictReader(file, delimiter=',')`